### PR TITLE
Obmin 463 optimization of slq requests to delete images 

### PR DESCRIPTION
**Pull request by Sershok**

### CODE REVIEW

File: src/test/java/space/obminyashka/items_exchange/rest/ImageControllerIntegrationTest.java [hunk 1]
```diff
@@ -146,12 +146,14 @@ void getByAdvertisementId_shouldReturn404WhenAdvertisementIsNotExist() throws Ex
     @WithMockUser("admin")
     @Test
     void deleteImages_shouldThrow400WhenImageIdNotExist() throws Exception {
+        when(advertisementService.isUserHasAdvertisementWithId(any(UUID.class), eq("admin"))).thenReturn(true);
```
Comment (IllyaZuew) [inline src/test/java/space/obminyashka/items_exchange/rest/ImageControllerIntegrationTest.java:149]: that can be moved to mocksInit method when(advertisementService.isUserHasAdvertisementWithId(advertisementId, "admin")).thenReturn(true);

File: src/main/java/space/obminyashka/items_exchange/rest/ImageController.java [hunk 2]
```diff
@@ -143,17 +142,14 @@ public void deleteImages(@Parameter(name = "advertisement_id", description = "ID
                              @Parameter(hidden = true) Authentication authentication)
             throws IllegalOperationException, IllegalIdentifierException {
 
-        final var existedAdvertisement = advertisementService.findByIdAndOwnerUsername(advertisementId, authentication.getName())
-                .orElseThrow(() -> new IllegalOperationException(getMessageSource(ResponseMessagesHandler.ValidationMessage.USER_NOT_OWNER)));
+        if (!advertisementService.isUserHasAdvertisementWithId(advertisementId, authentication.getName())) {
+            throw new IllegalOperationException(getMessageSource(ResponseMessagesHandler.ValidationMessage.USER_NOT_OWNER));
```
Comment (IllyaZuew) [inline src/main/java/space/obminyashka/items_exchange/rest/ImageController.java:146]: you should remove full path in Message parameter

File: src/main/java/space/obminyashka/items_exchange/rest/ImageController.java [hunk 3]
```diff
@@ -143,17 +142,14 @@ public void deleteImages(@Parameter(name = "advertisement_id", description = "ID
...
+        if (imageService.existAllById(imageIdList, advertisementId)) {
             imageService.removeById(imageIdList);
         } else {
-            final var existedImageIds = existedAdvertisement.getImages().stream()
-                    .map(Image::getId)
-                    .toList();
-            imageIdList.removeAll(existedImageIds);
+            imageIdList.removeAll(imageService.getImagesIdByAdvertisementId(advertisementId));
             throw new IllegalIdentifierException(getParametrizedMessageSource(
                     ResponseMessagesHandler.ExceptionMessage.IMAGE_NOT_EXISTED_ID, imageIdList));
```
Comment (IllyaZuew) [inline src/main/java/space/obminyashka/items_exchange/rest/ImageController.java:154]: you should remove full path in Message parameter

File: src/main/java/space/obminyashka/items_exchange/repository/ImageRepository.java [hunk 4]
```diff
@@ -17,4 +18,7 @@ public interface ImageRepository extends JpaRepository<Image, UUID> {
     boolean existsAllByIdInAndAdvertisement_Id(List<UUID> id, UUID advertisementId);
 
     int countImageByAdvertisement_Id(UUID id);
+
+    @Query("select i.id from Image i where i.advertisement.id =:advertisementId")
+    List<UUID> getImagesIdByAdvertisementId(UUID advertisementId);
```
Comment (IllyaZuew) [inline src/main/java/space/obminyashka/items_exchange/repository/ImageRepository.java:23]: try to write it without @Query

**Review by IllyaZuew — APPROVED** [review]

**Review by ennnbaev — APPROVED** [review]

**Review by demus-n — APPROVED** [review]

**Review by ImAndVlad — APPROVED** [review]

**Pull request merged**